server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -246,6 +265,12 @@ public void search(
 
     @Override
     protected void search(List<LeafReaderContext> leaves, Weight weight, Collector collector) throws IOException {
+        if (reverseLeafReaderContexts) {
+            // reverse the segment search order if this flag is true.
+            // We need to make a copy, so we don't corrupt DirectoryReader leaves order
+            leaves = new ArrayList<>(leaves);
+            Collections.reverse(leaves);
+        }
         for (LeafReaderContext ctx : leaves) { // search each subreader
```

Comment:
Is it possible iterate on the existing list in reverse order based on the boolean flag? That will prevent the additional allocation and reverse operation.

Reply:
Yes. makes sense